### DISCUSSION TITLE
GitHub login status gets lost after restarting of code-server

### DISCUSSION THREAD

**Discussion by karuboniru** [coder/code-server#5052] [Q&A]
GitHub login status gets lost after a restart of server (in extensions like copilot). Can this be somehow resolved?

**Comment by karuboniru** [comment]
noticed 
```
Switching to using in-memory credential store instead because Keytar failed to load: Cannot find module 'keytar'
``` 
in log, is this related?

**Reply by jsjoeio** [reply]
I don't know if it's related 🤔 I don't think VS Code was using `keytar` previously so it was marked as an optional dependency. That should change in the next version.

RE: losing GitHub login status. How are you using code-server? I'm not sure where they store your login state.

**Reply by karuboniru** [reply]
I doubt just adding keytar back will help or not. I don't think there is a daemon on my headless machine providing the storage backend. 

While in [this message](https://github.com/coder/code-server/pull/2961#issuecomment-806576796) the credential information should be stored by browser (local storage)? While it did not work for me, is this a browser related problem?

And seems that there is a config option called `github-auth`, but which token shoud I put in? I tried github personal token and a token grabbed from my desktop libsecret storage, while none of them worked

**Reply by karuboniru** [reply]
Tried the official instance of vscode on web (vscode.dev), it is storing some credential in my browser's local storage.

**Reply by lg-nineoneninethreenine** [reply]
Hmm will switching to localStorage help?

**Reply by jsjoeio** [reply]
Hmm... you should be able to use at PAT (personal access token) and that should work. 

> Tried the official instance of vscode on web (vscode.dev), it is storing some credential in my browser's local storage.

Does that mean vscode.dev has the same issue?

**Reply by karuboniru** [reply]
> Hmm... you should be able to use at PAT (personal access token) and that should work.

Tried the PAT (ghp_xxx) token, tried both way to pass the token to code-server (via `GITHUB_TOKEN` env or `github-auth` in config file). Still have to re-login after restarting of code-server daemon.

And the token is still marked as `Never used` in GitHub page, does it mean I am using the token in a wrong way?

> Does that mean vscode.dev has the same issue?

No, when on vscode.dev, a local storage entry `credentials.provider` is created in my browser and it not the case when in code-server.

**Reply by karuboniru** [reply]
And just some in-case information, I am using the current latest version `4.2.0`, behind reverse proxy nginx in a subpath. Client is FireFox 98.0. Code-server is installed via the rpm package in github release page

Nginx configuration is: 
```
location /code/ {
  proxy_pass http://127.0.0.1:3000/;
  proxy_set_header Host $host;
  proxy_set_header Upgrade $http_upgrade;
  proxy_set_header Connection upgrade;
  proxy_set_header Accept-Encoding gzip;
} 
```

**Reply by karuboniru** [reply]
> Hmm will switching to localStorage help?

I suppose the switch should be done automatically, or is there any to force that?

**Reply by lg-nineoneninethreenine** [reply]
Possibly. It's TypeScript, so it could be easy to add a token to localStorage. The only thing is that we should call it when we start the session.

**Reply by jsjoeio** [reply]
If you remove the reverse proxy from the equation, does this issue still happen?

**Reply by karuboniru** [reply]
> If you remove the reverse proxy from the equation, does this issue still happen?

no luck, still need to re-login after a restart

**Reply by jsjoeio** [reply]
Sounds like s bug then. Do you mind opening a bug report?
